WordPress/Classes/ViewRelated/Stats/Shared Views/Date Chooser/SiteStatsTableHeaderView.swift

```diff
@@ -0,0 +1,82 @@
+import UIKit
+
+class SiteStatsTableHeaderView: UITableViewHeaderFooterView, NibLoadable {
+
+    // MARK: - Properties
+
+    @IBOutlet weak var dateLabel: UILabel!
+    @IBOutlet weak var backArrow: UIImageView!
+    @IBOutlet weak var forwardArrow: UIImageView!
+    @IBOutlet weak var bottomSeparatorLine: UIView!
+
+    static let height: CGFloat = 44
+    private typealias Style = WPStyleGuide.Stats
+
+
+    // MARK: - View
+
+    override func awakeFromNib() {
+        applyStyles()
+    }
+
+    func configure(date: Date?, period: StatsPeriodUnit?) {
+
+        guard let date = date, let period = period else {
+            return
+        }
+
+        dateLabel.text = displayDateFor(date: date, period: period)
+    }
+
+}
+
+private extension SiteStatsTableHeaderView {
+
+    func applyStyles() {
+        Style.configureLabelAsCellRowTitle(dateLabel)
+        Style.configureViewAsSeparator(bottomSeparatorLine)
+        backArrow.image = Style.imageForGridiconType(.chevronLeft, withTint: .darkGrey)
+        forwardArrow.image = Style.imageForGridiconType(.chevronRight)
+    }
+
+    func displayDateFor(date: Date, period: StatsPeriodUnit) -> String {
+        let dateFormatter = DateFormatter()
+        dateFormatter.setLocalizedDateFormatFromTemplate(period.dateFormatTemplate)
+
+        switch period {
+        case .day, .month, .year:
+            return dateFormatter.string(from: date)
+        case .week:
+            // Week is Monday - Sunday
+            var calendar = Calendar(identifier: .iso8601)
+            calendar.timeZone = .autoupdatingCurrent
+
+            let weekStart = calendar.date(from: calendar.dateComponents([.yearForWeekOfYear, .weekOfYear], from: date))!
```

Comment:
Hey @stevebaranski - I just noticed I forgot to remove the `!`. If it's OK with you, I'll fix it in my next PR, which I'm working on now.

Reply:
I'll allow it. 😄

Reply:
It doesn't appear like we have anything quite like this in `WordPressShared`. Perhaps after we ship, we can identify candidate logic like this for promotion to that module?

Reply:
Sure. You're right, later is probably better since I'm still modifying this class. But yea, we can do that.